### DISCUSSION TITLE
Poetry develop?

### DISCUSSION THREAD

**Discussion by iamgp** [python-poetry/poetry#3784] [Q&A]
I have two directories using poetry to manage virtual environments, dependencies etc:

```
/dirA/mypackage
/dirB
```

In dirB, I can add dirA as a dependency by `poetry add ../dirA`. If I import the package and print out the version  it works:
```python
import mypackage
print(mypackage.__version__) # returns 0.1
```

From  perusing [this discussion](https://github.com/python-poetry/poetry/discussions/1135) and the docs, I assumed that dirA/mypackage would be editable, i.e. if I bump the version number in dirA/mypackage/__version__.py to 0.2, then in dirB this should reflect the change.

However, it doesn't. 
```python
import mypackage
print(mypackage.__version__) # STILL returns 0.1
```

What can I do if I would like to do this? Something akin to python setup.py develop in poetry speak?

Thanks for your help!

**Comment by finswimmer** [comment]
Hello,

since poetry 1.1 path dependencies are not installed in editable mode by default anymore (see [docs](https://python-poetry.org/docs/dependency-specification/#path-dependencies)). You have to specify it explicit in the `pyproject.toml`, e.g. :

```
[tool.poetry.dependencies]
my-package = { path = "../my-package/", develop = true }
```

Don't forget to run `poetry lock` and `poetry install` after you've edit the `pyproject.toml`.

fin swimmer

**Reply by iamgp** [reply]
Hey fin swimmer,

Thanks! I did see that but for some reason thought that made mypackage's dependencies editable, my mistake. 

I just wanted to check something though with this -

If I am in `dirB` and have installed `mypackage` as a dependency as you described, everything works correctly if i go `poetry run python main.py` or `poetry shell` then `import mypackage; print(mypackage.__version__)`, but if I run `poetry shell` then `python main.py` it can't find the `mypackage` dependency... Is that expected behaviour?

**Reply by iamgp** [reply]
Furthermore, in Vscode, even if i select the correct virtualenv interpreter, the IDE can't find the dependency either...

**Reply by michalszelagsonos** [reply]
This workflow seems to be broken on Ubuntu 18.04 and 20.04 right now on `1.1.6`. 
- do the manual edit in `pyproject.toml`
- run `poetry lock`
- run `poetry install`
Poetry shows me that my editable package is installed but the changes do not take effect and pip is certainly not showing that the package is integrated in venv as editable.  I really hope this is fixed in `1.2` since editable installs are pretty important in my org, given we have several repos that people need to work on at the same time.
